src/toolbar/toolbar-button.ts

```diff
@@ -5,13 +5,18 @@ export function ToolbarButton(
   editor: MinidocToolbarEditor,
   { label, isActive, html, run }: Pick<MinidocToolbarAction, 'label' | 'isActive' | 'html' | 'run'>,
 ) {
+  let lastActive: Element | null;
   const btn = h('button.minidoc-toolbar-btn', {
     refreshState:
       isActive &&
       ((editor: MinidocToolbarEditor) =>
         btn.classList.toggle('minidoc-toolbar-btn-active', isActive(editor))),
+    onmousedown: () => {
+      lastActive = document.activeElement;
+    },
     onclick() {
       run(editor);
+      (lastActive as HTMLElement)?.focus();
```

Comment:
My hunch is that you could replace this PR with this:

```js
editor.root.focus();
```

`document.activeElement` could point anywhere depending on what was focused prior to clicking the toolbar. (e.g. if some other text input in the form was focused).

Reply:
Oh, right. That wouldn't work to fix https://github.com/morebetterlabs/ruzuku-v2/issues/3068 as the title is a simple `input` element in that case. I guess manipulating a regular input from the minidoc toolbar is not ideal but it was the simplest approach.

Reply:
Ah. I see. I think in that case, we should fix that in the Ruzuku source itself. We can add a special click / whatever handler to that specific action and take the appropriate steps afterwards since it's not actually a minidoc action.

Reply:
👍 I removed the logic from minidoc and will move it to Ruzuku. I modified the PR so we are passing `onmousedown` events to the toolbar buttons.